data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -76,4 +83,81 @@ public Environment getEnvironment(
 
     return environmentMapper.toEnvironment(environmentDDBRecord);
   }
+
+  @Override
+  public Environment createEnvironmentTargetVersion(final Environment environment)
+      throws StorageException, EnvironmentExistsException {
+
+    final EnvironmentTargetVersionDDBRecord environmentTargetVersionDDBRecord =
+        environmentMapper.toEnvironmentTargetVersionDDBRecord(environment);
+
+    try {
+      dynamoDBMapper.save(environmentTargetVersionDDBRecord);
+    } catch (final AmazonServiceException e) {
+      throw new StorageException(
+          String.format(
+              "Could not save record with environment id %s", environment.getEnvironmentId()),
+          e);
+    }
+    return environmentMapper.toEnvironment(environmentTargetVersionDDBRecord);
+  }
+
+  @Override
+  public Environment getEnvironmentTargetVersion(String environmentId)
+      throws StorageException, EnvironmentNotFoundException {
+
+    final EnvironmentTargetVersionDDBRecord environmentTargetVersionDDBRecord;
+    try {
+      environmentTargetVersionDDBRecord =
+          dynamoDBMapper.load(EnvironmentTargetVersionDDBRecord.withHashKey(environmentId));
+    } catch (final AmazonServiceException e) {
+      throw new StorageException(
+          String.format("Could not load record with environment id %s", environmentId), e);
+    }
+
+    if (environmentTargetVersionDDBRecord == null) {
+      throw new EnvironmentNotFoundException(
+          String.format("Could not find environment with id %s", environmentId));
+    }
+
+    return environmentMapper.toEnvironment(environmentTargetVersionDDBRecord);
+  }
+
+  @Override
+  public List<String> listClusters() throws StorageException {
+    try {
+      List<EnvironmentTargetVersionDDBRecord> scanResult =
+          dynamoDBMapper.scan(
+              EnvironmentTargetVersionDDBRecord.class, new DynamoDBScanExpression());
+      //TODO: handle pagination. paginatedlist returns more elements on demand does stream accomplish that?
```

Comment:
`stream` does it on-demand, but `collect` is a terminal stream operation, so it will load everything, eventually.

If we really want to eager-load all results before mapping over them, there's `PaginatedList.loadAllResults()`, but I can't see much benefit to that.

Reply:
so collect will load everything? good to know, we'll have to add an integration test for that